benchmarks/persistence/persistence.py

```diff
@@ -0,0 +1,228 @@
+    """The boto3 session object. This will load the respective credentials
+    from the environment variables within the container."""
+
+    @_bucket_name.default
+    def _default_bucket_name(self) -> str:
+        """Get the bucket name from the environment variables."""
+        if "BAYBE_PERFORMANCE_PERSISTENCE_PATH" not in os.environ:
+            raise ValueError(
+                "No S3 bucket name provided. Please provide the "
+                + "bucket name by setting the environment variable "
+                + "BAYBE_PERFORMANCE_PERSISTENCE_PATH."
+            )
+        return os.environ["BAYBE_PERFORMANCE_PERSISTENCE_PATH"]
+
+    def write_object(self, path_constructor: PathConstructor, object: dict) -> None:
+        """Store a JSON serializable dict according to the path.
+
+        This method will store an JSON serializable dict in an S3 bucket.
+        The S3-key of the Java Script Notation Object will be created by
+        the path object. I the key already exists, it will be overwritten.
+
+        Args:
+            path_constructor: The PathConstructor to create the path for the object.
+            object: The object to be persisted.
+        """
+        client = self._object_session.client("s3")
+
+        client.put_object(
+            Bucket=self._bucket_name,
+            Key=path_constructor.get_path(strategy=PathStrategy.HIERARCHICAL),
+            Body=json.dumps(object),
+            ContentType="application/json",
+        )
+
+
+@define
+class LocalFileSystemPersistenceHandler:
+    """Class for persisting JSON serializable dicts locally."""
+
+    folder_path_prefix: str = field(validator=instance_of(str), default="./")
+    """The prefix of the folder path where the results are stored.
+    The filename will be created automatically and create or override the
+    file under this path. The file path must exist."""
+
+    def write_object(self, path_constructor: PathConstructor, object: dict) -> None:
+        """Store a JSON serializable dict according to the path.
+
+        This method will store an JSON serializable dict in a local file system.
+        If the respective file exists, it will be overwritten.
+
+        Args:
+            path_constructor: The PathConstructor to create the path for the object.
+            object: The object to be persisted.
+
+        Raises:
+            FileNotFoundError: If the folder path prefix does not exist.
+        """
+        folder_path = Path(self.folder_path_prefix)
+        if not folder_path.exists():
+            raise FileNotFoundError(
+                f"The folder path {folder_path.resolve()} does not exist."
+            )
+        path_object = folder_path.joinpath(
+            Path(path_constructor.get_path(strategy=PathStrategy.FLAT))
+        )
+        with open(path_object.resolve(), "w") as file:
+            json.dump(object, file)
+
+
+def persister_factory() -> PersistenceHandlingInterface:
```

Comment:
`make_object_writer` is more consistent with our style (I'd rather use the term `Factory` when using an object-oriented approach)